src/styles/marg.js

```diff
@@ -10,47 +10,37 @@ const marg = ( props ) => {
     // see cyverse-ui/src/style/README.md
     //
 
+    const returnValue = (value, sizes) => {
+        if (typeof value == "number") {
+            return value === 0 ? 0 : sizes[value -1]
+        } else { 
+            return value
+        }
+    };
+
     return Object.keys(props).reduce((style, prop) => {
         let value = props[prop];
 
         let cssField;
         switch (prop) {
             case "mr":
-                style.marginRight = typeof value == "number"
-                    ? sizes[value - 1]
-                    : value
+                style.marginRight = returnValue(value, sizes);
                 break;
             case "mb":
-                style.marginBottom = typeof value == "number"
-                    ? sizes[value - 1]
-                    : value
+                style.marginBottom = returnValue(value, sizes);
                 break;
             case "ml":
-                style.marginLeft = typeof value == "number"
-                    ? sizes[value - 1]
-                    : value
+                style.marginLeft = returnValue(value, sizes);
                 break;
             case "ms":
-                style.marginRight = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginLeft = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
+                style.marginRight = returnValue(value, sizes);
+                style.marginLeft = returnValue(value, sizes);
                 break;
             case "m":
-                style.marginTop = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginRight = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginBottom = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginLeft = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
+                style.marginTop = returnValue(value, sizes);
+                style.marginRight = returnValue(value, sizes);
+                style.marginBottom = returnValue(value, sizes);
+                style.marginLeft = returnValue(value, sizes);
```

Comment:
Havn't tested this, but this is a cleaned up version (more terse)
```diff
return Object.keys(props).reduce((style, rule) => {
    let metric = props[rule];
    let margin = metric;
    if (typeof metric == "number") {
        margin = metric === 0 ? 0 : sizes[metric - 1];
    }
    switch (rule) {
        case "m":
        case "mv":
        case "mt":
            style.marginTop = margin;
        case "m":
        case "mv":
        case "mb":
            style.marginBottom = margin;
        case "m":
        case "ms":
        case "mr":
            style.marginRight = margin;
        case "m":
        case "ms":
        case "ml":
            style.marginLeft = margin;
    }

    return style;

}, {});
```

Here is a less terse, version:
```
return Object.keys(props).reduce((style, rule) => {
    let metric = props[rule];
    let margin = metric;
    if (typeof metric == "number") {
        margin = metric === 0 ? 0 : sizes[metric - 1];
    }
    switch (rule) {
        case "mt":
            style.marginTop = margin;
            break;
        case "mr":
            style.marginRight = margin;
            break;
        case "mb":
            style.marginBottom = margin;
            break;
        case "ml":
            style.marginLeft = margin;
            break;
        case "ms":
            style.marginRight = style.marginLeft = margin;
            break;
        case "mv":
            style.marginTop = style.marginBottom = margin;
            break;                 
        case "m":                  
            style.marginTop = style.marginRight = style.marginBottom = style.marginLeft = margin;
            break;                 
    }                              

    return style;

}, {});
```